exercises/matrix/src/example/java/Matrix.java

```diff
@@ -0,0 +1,35 @@
+public class Matrix {
+    private int[][] matrix;
+
+    public Matrix(String matrixAsString) {
+        String[] rows = matrixAsString.split("\\n");
+        matrix = new int[rows.length][];
+        for (int i = 0; i < rows.length; i++) {
+            String[] columnValues = rows[i].split(" ");
```

Comment:
Another FYI, not a required change: note that there are some "gotchas" when using `String.split` with a regex. [Here's a nice breakdown](http://chrononsystems.com/blog/hidden-evils-of-javas-stringsplit-and-stringr). If this was high-performance code, you'd probably want to compile the `Pattern` ahead of time, and split using that instead. I only learned about this a couple months back.